sdk/containerregistry/azure-containerregistry/azure/containerregistry/_generated/models/_models.py

```diff
@@ -312,257 +331,126 @@ class ArtifactManifestProperties(msrest.serialization.Model):
     :vartype registry_login_server: str
     :ivar repository_name: Repository name.
     :vartype repository_name: str
-    :ivar digest: Required. Manifest.
-    :vartype digest: str
-    :ivar size: Image size.
-    :vartype size: long
-    :ivar created_on: Required. Created time.
-    :vartype created_on: ~datetime.datetime
-    :ivar last_updated_on: Required. Last update time.
-    :vartype last_updated_on: ~datetime.datetime
-    :ivar architecture: CPU architecture. Known values are: "386", "amd64", "arm", "arm64", "mips",
-     "mipsle", "mips64", "mips64le", "ppc64", "ppc64le", "riscv64", "s390x", "wasm".
-    :vartype architecture: str or ~container_registry.models.ArtifactArchitecture
-    :ivar operating_system: Operating system. Known values are: "aix", "android", "darwin",
-     "dragonfly", "freebsd", "illumos", "ios", "js", "linux", "netbsd", "openbsd", "plan9",
-     "solaris", "windows".
-    :vartype operating_system: str or ~container_registry.models.ArtifactOperatingSystem
-    :ivar related_artifacts: List of artifacts that are referenced by this manifest list, with
-     information about the platform each supports.  This list will be empty if this is a leaf
-     manifest and not a manifest list.
-    :vartype related_artifacts: list[~container_registry.models.ArtifactManifestPlatform]
-    :ivar tags: A set of tags. List of tags.
-    :vartype tags: list[str]
-    :ivar can_delete: Delete enabled.
-    :vartype can_delete: bool
-    :ivar can_write: Write enabled.
-    :vartype can_write: bool
-    :ivar can_list: List enabled.
-    :vartype can_list: bool
-    :ivar can_read: Read enabled.
-    :vartype can_read: bool
+    :ivar manifest: Manifest attributes. Required.
+    :vartype manifest: ~container_registry.models.ManifestAttributesBase
     """
 
     _validation = {
-        'registry_login_server': {'readonly': True},
-        'repository_name': {'readonly': True},
-        'digest': {'required': True, 'readonly': True},
-        'size': {'readonly': True},
-        'created_on': {'required': True, 'readonly': True},
-        'last_updated_on': {'required': True, 'readonly': True},
-        'architecture': {'readonly': True},
-        'operating_system': {'readonly': True},
-        'related_artifacts': {'readonly': True},
-        'tags': {'readonly': True},
+        "registry_login_server": {"readonly": True},
+        "repository_name": {"readonly": True},
+        "manifest": {"required": True, "readonly": True},
```

Comment:
I'm confused, should manifest be readonly? It says required in the docstring. This is why we're seeing mypy error in container_registry_client.py L421 and 654

Reply:
Comparing with the code changes, manifest is `readonly and required`. @iscai-msft Can you help to confirm?

Reply:
We're documenting the ivar as required, but not offering it as input. Not sure why mypy would be mad here, it's not accepting it as input. What is the mypy error?

Reply:
`error: Argument 1 to "_from_generated" of "ArtifactManifestProperties" has incompatible type "None"; expected "ManifestAttributesBase"`